harness/tests/experiment/keras/test_callback.py

```diff
@@ -82,6 +97,12 @@ def on_train_begin(self, logs: Any) -> None:
         fourdigits = "%.4f" % weight
         self.events.append((f"after_train_begin:{fourdigits}", weight))
 
+    def on_train_batch_end(self, batch: int, logs: Any) -> None:
+        if not self.first_train_batch_end:
```

Comment:
why not make this "before_batch_end:batch"?

Reply:
That's what I had at first, but it would really just pollute error logs with a bunch of junk that nobody cares about when any of the tests fail.

Since we only ever care about the first train batch end, I decided to take the minimally-event-polluting approach.